web/components/templates/sessions/sessionDetails.tsx

```diff
@@ -41,13 +54,68 @@ const SessionDetails = ({
   setInterval,
 }: SessionDetailsProps) => {
   const router = useRouter();
+  const [lastUsedSession, setLastUsedSession] = useState<TSessions>();
+  const [oldestSession, setOldestSession] = useState<TSessions>();
+  const [totalCost, setTotalCost] = useState("0.0");
+
+  function calculateMetadata() {
+    if (sessions.length === 0) return;
+
+    setLastUsedSession(
+      sessions.reduce((latestSession, currentSession) => {
+        return new Date(currentSession.latest_request_created_at) >
+          new Date(latestSession.latest_request_created_at)
+          ? currentSession
+          : latestSession;
+      })
+    );
+
+    setTotalCost(
+      sessions.reduce((acc, session) => acc + session.total_cost, 0).toFixed(3)
+    );
+
+    setOldestSession(
+      sessions.reduce((latestSession, currentSession) => {
+        return new Date(currentSession.latest_request_created_at) >
+          new Date(latestSession.latest_request_created_at)
+          ? currentSession
+          : latestSession;
+      })
+    );
+  }
+
+  useEffect(() => {
+    calculateMetadata();
+  }, [sessions]);
```

Comment:
this is fine for now, but in the future, you don't need to put this all in a usEffect. useEffects, especially ones that are setting state will cause rerenders. You can just set the variables directly like 
```ts
const oldestSession =  sessions.reduce((latestSession, currentSession) => {
        return new Date(currentSession.latest_request_created_at) >
          new Date(latestSession.latest_request_created_at)
          ? currentSession
          : latestSession;
      })

```